src/Covenant/Internal/Type.hs

```diff
@@ -266,6 +272,33 @@ data BuiltinFlatT
       Show
     )
 
+naturalBaseFunctor :: DataDeclaration AbstractTy
+naturalBaseFunctor = DataDeclaration "Natural_F" count1 constrs SOP
+  where
+    constrs :: Vector (Constructor AbstractTy)
+    constrs =
+      [ Constructor "ZeroNatF" [],
```

Comment:
This should be `ZeroNat_F` if we're keeping with the naming scheme from the generator function. (Same for all the other constructors, I'm not going to tag all of them individually)

Reply:
OK, will fix.